Unnecessary D102 for property setter if getter has already a docstring

When a property has a setter and a getter, it should be enough to write a docstring for the getter only. After all, the property has only a single docstring. However, Ruff issues a D102 warning about the missing docstring for the setter.

Pydocstyle doesn't complain about this.

### Example 

file `property_with_setter.py`:
```python
class PropertyWithSetter:
    @property
    def foo(self) -> str:
        """Docstring for foo."""
        return "foo"

    @foo.setter
    def set_foo(self, value: str) -> None:
        pass
```

```
>ruff property_with_setter.py --isolated --select D102
property_with_setter.py:8:9: D102 Missing docstring in public method
Found 1 error(s).

>ruff --version
ruff 0.0.221
```

--- a/resources/test/fixtures/pydocstyle/setter.py
+++ b/resources/test/fixtures/pydocstyle/setter.py
@@ -1,0 +1,17 @@
+class PropertyWithSetter:
+    @property
+    def foo(self) -> str:
+        """Docstring for foo."""
+        return "foo"
+
+    @foo.setter
+    def foo(self, value: str) -> None:
+        pass
+
+    @foo.deleter
+    def foo(self):
+        pass
+
+    @foo
+    def foo(self, value: str) -> None:
+        pass

--- a/src/rules/pydocstyle/mod.rs
+++ b/src/rules/pydocstyle/mod.rs
@@ -17,7 +17,8 @@ mod tests {
 
     #[test_case(RuleCode::D100, Path::new("D.py"); "D100")]
     #[test_case(RuleCode::D101, Path::new("D.py"); "D101")]
-    #[test_case(RuleCode::D102, Path::new("D.py"); "D102")]
+    #[test_case(RuleCode::D102, Path::new("D.py"); "D102_0")]
+    #[test_case(RuleCode::D102, Path::new("setter.py"); "D102_1")]
     #[test_case(RuleCode::D103, Path::new("D.py"); "D103")]
     #[test_case(RuleCode::D104, Path::new("D.py"); "D104")]
     #[test_case(RuleCode::D105, Path::new("D.py"); "D105")]

--- a/src/visibility.rs
+++ b/src/visibility.rs
@@ -5,6 +5,7 @@ use std::path::Path;
 
 use rustpython_ast::{Expr, Stmt, StmtKind};
 
+use crate::ast::helpers::collect_call_path;
 use crate::checkers::ast::Checker;
 use crate::docstrings::definition::Documentable;
 
@@ -148,7 +149,29 @@ fn function_visibility(stmt: &Stmt) -> Visibility {
 
 fn method_visibility(stmt: &Stmt) -> Visibility {
     match &stmt.node {
-        StmtKind::FunctionDef { name, .. } | StmtKind::AsyncFunctionDef { name, .. } => {
+        StmtKind::FunctionDef {
+            name,
+            decorator_list,
+            ..
+        }
+        | StmtKind::AsyncFunctionDef {
+            name,
+            decorator_list,
+            ..
+        } => {
+            // Is this a setter or deleter?
+            if decorator_list.iter().any(|expr| {
+                let call_path = collect_call_path(expr);
+                print!("{call_path:?}");
+                if call_path.len() > 1 {
+                    call_path[0] == name
+                } else {
+                    false
+                }
+            }) {
+                return Visibility::Private;
+            }
+
             // Is the method non-private?
             if !name.starts_with('_') {
                 return Visibility::Public;

